components/CatItem.tsx

```diff
@@ -0,0 +1,262 @@
+import Modal from './ui/modal';
+import moment from 'moment';
+import catInfo from '../public/data.json';
+
+interface CatItemProps {
+  hasRescuerIdx?: boolean;
+  rescuerId?: number;
+  cat: Cat;
+  onClick(cat: Cat): void;
+  children: React.ReactChild;
+}
+
+export const CatListItem = ({
+  title,
+  value,
+}: {
+  title: string | React.ReactChild;
+  value: string | React.ReactChild;
+}) => {
+  return (
+    <div className="nft__meta_row">
+      <div className="nft__meta_title">{title}</div>
+      <div className="nft__meta_dot"></div>
+      <div className="nft__meta_value">{value}</div>
+    </div>
+  );
+};
+
+export const CatAdoptionDetail: React.FC<{ cat: Cat }> = ({ cat }) => {
+  const { provenance } = cat;
+  return (
+    <div className="adoption">
+      {provenance?.offerPrices && provenance?.offerPrices.length !== 0 && (
+        <div className="adoption__item">
+          <h3>Historical Offer Prices</h3>
+          <ul className="adoption__item_table">
+            {provenance.offerPrices
+              .sort(
+                (a, b) =>
+                  //@ts-ignore
+                  new Date(Number(a.timestamp) * 100) -
+                  //@ts-ignore
+                  new Date(Number(b.timestamp) * 100)
+              )
+              .map((item) => (
+                <li className="table-row" key={item.timestamp}>
+                  <span className="table-row-column">
+                    {moment(Number(item.timestamp) * 1000).format(
+                      'MMMM D YYYY h:mm'
+                    )}
+                  </span>
+                  <span className="table-row-column">
+                    {calculatePrice(item.price)}&nbsp;ETH
+                  </span>
+                </li>
+              ))}
+          </ul>
+        </div>
+      )}
+      {provenance?.requestPrices && provenance?.requestPrices?.length !== 0 && (
+        <div className="adoption__item">
+          <h3>Historical Request Prices</h3>
+          <ul className="adoption__item_table">
+            {/* @ts-ignore */}
+            {provenance.requestPrices
+              .sort(
+                (a, b) =>
+                  //@ts-ignore
+                  new Date(Number(a.timestamp) * 100) -
+                  //@ts-ignore
+                  new Date(Number(b.timestamp) * 100)
+              )
+              .map((item) => (
+                <li className="table-row" key={item.timestamp}>
+                  <span className="table-row-column">
+                    {moment(Number(item.timestamp) * 1000).format(
+                      'MMMM D YYYY h:mm'
+                    )}
+                  </span>
+                  <span className="table-row-column">
+                    {calculatePrice(item.price)}&nbsp;ETH
+                  </span>
+                </li>
+              ))}
+          </ul>
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const CatAdoptionRequest: React.FC<{ cat: Cat }> = ({ cat }) => {
```

Comment:
let's define the props: `type CatProp = {cat: Cat}` and use it everywhere, instead of rewriting the same code in props of other components